platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -131,13 +138,32 @@ private void doRollback(SolrClient solrClient, String type) {
   }
 
   @Override
-  // Returned Map will have suffixes in the key names - client is responsible for handling them
+  /**
+   * {@inheritDoc} Returned Map will have suffixes in the key names - client is responsible for
+   * handling them
+   */
   public List<Map<String, Object>> get(String type, String cql) throws PersistenceException {
+    return get(type, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+  }
+
+  @Override
+  public List<Map<String, Object>> get(String type, String cql, int startIndex, int pageSize)
+      throws PersistenceException {
     if (StringUtils.isBlank(type)) {
       throw new PersistenceException(
           "The type of object(s) to retrieve must be non-null and not blank, e.g., notification, metacard, etc.");
     }
 
+    if (startIndex < 0) {
+      throw new IllegalArgumentException("The start index must be nonnegative.");
+    }
+
+    if (pageSize <= 0 || pageSize > MAX_PAGE_SIZE) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The page size must be greater than 0 and less than or equal to %d.", MAX_PAGE_SIZE));
```

Comment:
✏️ For the base case of combining just two strings, simple string concatenation is preferred. Beyond that, a format string or builder is desirable due to various efficiency tradeoffs.

Reply:
this would be 3 concating 3 items together though `"initial text" + MAX_PAGE_SIZE _ + "."`

Reply:
and i don't think we say anywhere we prefer simple string concat over string format -- just that we **do** prefer string format when it helps legibility because of many items.